snap/local/matrix-ircd-wrapper

```diff
@@ -0,0 +1,25 @@
+#!/bin/bash
+
+if [ -f "${SNAP_COMMON}/matrix-ircd.env" ]; then
+    source "${SNAP_COMMON}/matrix-ircd.env"
+else
+    echo "No matrix-ircd.env file found in ${SNAP_COMMON}"
+    exit 1
+fi
+
+ARGS=""
+
+if [ ! -z "$MATRIX_IRCD_HOMESERVER" ]; then
+    ARGS="${ARGS} --url ${MATRIX_IRCD_HOMESERVER}"
+fi
+
+if [ ! -z "$MATRIX_IRCD_BIND" ]; then
+    ARGS="${ARGS} --bind ${MATRIX_IRCD_BIND}"
+fi
+
+if [ ! -z "$MATRIX_IRCD_PASSWORD" ] && [ ! -z "$MATRIX_IRCD_PKCS12" ]; then
+    ARGS="${ARGS} --password ${MATRIX_IRCD_PASSWORD} --pkcs12 ${SNAP_COMMON}/${MATRIX_IRCD_PKCS12}"
+fi
```

Comment:
I agree, I should document these in the README - and if you agree with the approach I am about to describe, I'll do that - but there may be a different way you wish to handle this, so I'll explain - 

These options are not snapcraft related, but rather environment variables. They are simply being sources from a file to create a rudimentary "configuration file" - as this script is being run as a systemd service, and without a configuration file, it is very difficult to change the values being passed to the executable for matrix-ircd (such as homeserver URL, etc.).

So, the snap registers a daemon, for matrix-ircd, but instead of running matrix-ircd directly, it runs this script, which sources `${SNAP_COMMON}/matrix-ircd.env` - which is intended to include the above variables. `SNAP_COMMON` itself is a snap-provided environment variables - which points to a special folder which gets created for each snap installed on a user's system (for example, `/var/snap/matrix-ircd/common` for a snap installed system-wide). This folder is important for snaps, as it is a folder that persists between snap upgrades, and is bind-mounted between the snap's isolated mount-namespace (isolated filesystem) and the host filesystem, so it is an easy place to put data you want the user to be able to edit, which will be saved between updates of the package.

One other approach would be to add support for loading a proper configuration file for matrix-ircd, in addition to supporting the current commandline parameters. That would invalidate the need for this wrapper script and the environment file - if a parameter was added to specify where the config file was stored, it could be pointed at `$SNAP_COMMON/matrix-ircd.yaml` for example, and the documentation could just tell snap users to put their config in `$SNAP_COMMON/matrix-ircd.yaml` and matrix-ircd's snap packaging could configure the systemd service to look for the config file there by default - so configuration for snap installs would just be editing one known file, which is pretty easy.

Let me know your thoughts on the approach, I'm happy to change how this is handled.

Reply:
I think the nicest solution would be the config file, but then we also should think about how to run multiple daemonized instances. 

So for now I think it's fine when you simply document the options in the readme :)

Reply:
Thanks again for the review, and sorry for the delay - I've updated the branch to include a snap section that details using and building the snap, and the options, as well as where to put the environment file.